File: src/main/java/com/autosale/shop/service/UserSessionService.java
package com.autosale.shop.service;

import com.autosale.shop.model.User;
import com.autosale.shop.model.UserSession;

import java.util.Optional;

public interface UserSessionService {
    UserSession createSession(User user);

    boolean terminateSession(int id);

    Optional<UserSession> getSession(User user);
}

File: src/main/java/com/autosale/shop/service/UserSessionService.kt
package com.autosale.shop.service

import com.autosale.shop.model.User
import com.autosale.shop.model.UserSession

interface UserSessionService {
    fun createSession(user: User): UserSession
    fun terminateSession(id: Int): Boolean
    fun getSession(user: User): UserSession
}
